python/BioSimSpace/Protocol/_AToM.py

```diff
@@ -0,0 +1,3287 @@
+                )
+            # convert the "displacement" key back to a vector
+            d = sdata["displacement"]
+            displacement = _Vector(*d)
+            sdata["displacement"] = displacement
+            self._system_data = sdata
+
+        elif system is not None and data is not None:
+            _warnings.warn(
+                "Both 'system' and 'data' were passed. Using 'data' and ignoring data from 'system'."
+            )
+
+        # Store the AToM system.
+        if isinstance(data, dict):
+            self._system_data = data
+        elif data is not None:
+            raise TypeError("'data' must be of type 'dict'")
+
+        # Whether or not to use alignment restraints.
+        self._setCoreAlignment(core_alignment)
+
+        # Whether or not to use the CMCM restraint.
+        self._setCMCMRestraint(CMCM_restraint)
+
+        # Store the width of the coordinate retraint.
+        self._setPosRestWidth(pos_rest_width)
+
+        # Store the align_kf_sep value.
+        self._setAlignKfSep(align_kf_sep)
+
+        # Store the align_k_theta value.
+        self._setAlignKTheta(align_k_theta)
+
+        # Store the align_k_psi value.
+        self._setAlignKPsi(align_k_psi)
+
+        # Store the SC_umax value.
+        self._setSCUmax(SC_umax)
+
+        # Store the SC_u0 value.
+        self._setSCU0(SC_u0)
+
+        # Store the SC_a value.
+        self._setSCa(SC_a)
+
+        # Store cm_kf value.
+        self._setCMKf(cm_kf)
+
+        # Store cm_tol value.
+        self._setCMTol(cm_tol)
+
+        # Set the postition restraint.
+        _PositionRestraintMixin.__init__(self, restraint, force_constant)
+
+    def __str__(self):
+        d = self._getData()
+        """Return a string representation of the protocol."""
+        string = "<BioSimSpace.Protocol.AToM>: "
+        string += "timestep=%s " % self.getTimeStep()
+        string += ", runtime=%s " % self.getRunTime()
+        string += ", temperature=%s " % self._getTemperature()
+        if self._pressure is not None:
+            string += ", pressure=%s, " % self._getPressure()
+        string += ", lambda1=%s " % self._getLambda1()
+        string += ", lambda2=%s " % self._getLambda2()
+        string += ", ligand1 core atoms=%s" % d["ligand1_rigid_core"]
+        string += ", ligand2 core atoms=%s" % d["ligand2_rigid_core"]
+        string += ", report_interval=%s " % self.getReportInterval()
+        string += ", restart_interval=%s " % self.getRestartInterval()
+        string += ">"
+
+        return string
+
+    def __repr__(self):
+        """Return a string showing how to instantiate the object."""
+        return self.__str__()
+
+    def _getData(self):
+        """
+        Return the AToM data dictionary.
+
+        Returns
+        -------
+
+        data : dict
+            The AToM data dictionary.
+        """
+        return self._system_data
+
+    def _getCoreAlignment(self):
```

Comment:
Can the setters and getters be public, as is the case for the other protocols? (That is, don't start with an `_`.) The use case was that sometimes protocols need to get modified further down the chain, or by the user, so having exposed setters (that validate things in a self-consistent way) makes it easier to avoid the user breaking things, e.g. by trying to set the attribute directly. I imagine that the AToM functionality is pretty self contained, so this is unlikely needed,  but it would be good to make things consistent.

Reply:
Setters and getters are now public. There are two exceptions to this:
- `_set_current_index` and `_get_current_index` in the annealing protocol. This is used internally to set the initial values of the lambda-dependent variables, which should not be touched by the user. 
- `_set_lambda_values` and `_get_lambda_values` within the production protocol. These are used to make the _master_ lambda value, but don't change anything internally (they're just for file naming), allowing the user to set these would just cause confusion. It should also be noted that the production protocol has its own versions of `set_current_index` and `get_current_index`. These are something that could very easily break things should the user not set them properly, and they might need to be set to private.